### ISSUE TITLE
[BUG] sof-test output is messy when run test case locally

### ISSUE THREAD

**Issue by aiChaoSONG** [thesofproject/sof-test#1151]
**Describe the bug**
When run test case locally, the output of the test case is messy. Log doesn't start from the beginning of a line. This makes the log hard to follow.
![Screenshot from 2024-01-16 16-31-02](https://github.com/thesofproject/sof-test/assets/52959957/0716df4b-6818-4cb6-94e6-9bb948a2f356)

**To Reproduce**
Just run a test case, say check-playback.sh on any device, and check the output.

EDIT, duplicate of:
- #945

**Comment by aiChaoSONG** [comment]
`sudo -E "${mtraceCmd[@]}" >& "$clogfile" & `  this will mitigate the issue a little bit.

**Comment by marc-hb** [comment]
Let's leave open as low priority because it's just a workaround and we have no idea why it works.

**Comment by marc-hb** [comment]
I've just observed the same effect in a totally different (build) context. I think it happens when multiple processes write to the same terminal concurrently and one of them uses control codes (e.g. color). I suspect control codes are not "atomic" which means another problem can insert characters in the middle and break them.

Maybe the workaround #1154 disables control codes by hiding the terminal.

This would also explains why CI results are not affected: they don't use a terminal.

**Comment by marc-hb** [comment]
I  got tired of this and I tested further. This is a problem only when using `sudo + &`
```
/bin/sudo    sleep 1 &   # garbage
/bin/sudo -b sleep 1     # clean
/bin/sudo    sleep 1     # clean
/bin/sudo -C sleep 1     # clean
```

```diff
--- a/case-lib/lib.sh
+++ b/case-lib/lib.sh
@@ -463,7 +463,7 @@ func_sof_logger_collect()
     dlogi "Starting ${loggerCmd[*]} > $logfile &"
     # Cleaned up by func_exit_handler() in hijack.sh
     # shellcheck disable=SC2024
-    sudo "${loggerCmd[@]}" > "$logfile" &
+    /bin/sudo sleep 1 &
 }
```
